_dev/src/ts/components/ModalContainer.ts

```diff
@@ -0,0 +1,57 @@
+import Hydration from '../utils/Hydration';
+
+export default class ModalContainer {
+  public static readonly cancelEvent = 'cancel';
+  public static readonly okEvent = 'ok';
+
+  public static readonly containerId = 'ua_modal';
+
+  public init(): void {
+    // Avoid reattaching the same event listener several times.
+    this.unset();
+    this.modalContainer.addEventListener(Hydration.hydrationEventName, this.displayModal);
+    this.modalContainer.addEventListener('click', this.onClick);
+    this.modalContainer.addEventListener(ModalContainer.cancelEvent, this.closeModal);
+    this.modalContainer.addEventListener(ModalContainer.okEvent, this.closeModal);
+  }
+
+  public unset(): void {
+    this.modalContainer.removeEventListener(Hydration.hydrationEventName, this.displayModal);
+    this.modalContainer.removeEventListener('click', this.onClick);
+    this.modalContainer.removeEventListener(ModalContainer.cancelEvent, this.closeModal);
+    this.modalContainer.removeEventListener(ModalContainer.okEvent, this.closeModal);
+  }
+
+  private displayModal(): void {
+    $(
+      document.getElementById(ModalContainer.containerId)?.getElementsByClassName('modal') || []
+    ).modal('show');
+  }
+
+  private onClick(ev: Event) {
+    const target = ev.target ? (ev.target as HTMLElement) : null;
+    const modal = target?.closest('.modal');
+
+    if (modal && target?.closest("[data-dismiss='modal']")) {
+      modal.dispatchEvent(new Event(ModalContainer.cancelEvent, { bubbles: true }));
+    } else if (modal && target?.closest(".modal-footer button:not([data-dismiss='modal'])")) {
+      modal.dispatchEvent(new Event(ModalContainer.okEvent, { bubbles: true }));
+    }
```

Comment:
```suggestion
    if (modal) {
      const dismissTarget = target?.closest("[data-dismiss='modal']");
      const actionTarget = target?.closest(".modal-footer button:not([data-dismiss='modal'])");

    if (dismissTarget) {
      modal.dispatchEvent(new Event(ModalContainer.cancelEvent, { bubbles: true }));
    } else if (actionTarget) {
      modal.dispatchEvent(new Event(ModalContainer.okEvent, { bubbles: true }));
    }
}
```

Reply:
I don't think we need to set variables as they are only used for the conditions.

What about

```suggestion
    if (modal) {
      if (target?.closest("[data-dismiss='modal']")) {
        modal.dispatchEvent(new Event(ModalContainer.cancelEvent, { bubbles: true }));
      } else if (target?.closest(".modal-footer button:not([data-dismiss='modal'])")) {
        modal.dispatchEvent(new Event(ModalContainer.okEvent, { bubbles: true }));
      }
    }
```

Reply:
the variables are present in order to improve the understanding of the code. It is possible to do without